x-pack/plugins/endpoint/server/index_pattern.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Logger, LoggerFactory, RequestHandlerContext } from 'kibana/server';
+import { ESIndexPatternService } from '../../ingest_manager/server';
+import { EndpointAppConstants } from '../common/types';
+
+export interface IndexPatternRetriever {
+  getIndexPattern(ctx: RequestHandlerContext, datasetPath: string): Promise<string>;
+  getEventIndexPattern(ctx: RequestHandlerContext): Promise<string>;
+  getMetadataIndexPattern(ctx: RequestHandlerContext): Promise<string>;
+}
+
+export class IngestIndexPatternRetriever implements IndexPatternRetriever {
+  private static endpointPackageName = 'endpoint';
+  private static metadataDataset = 'metadata';
+  private readonly log: Logger;
+  constructor(private readonly service: ESIndexPatternService, loggerFactory: LoggerFactory) {
+    this.log = loggerFactory.get('index-pattern-retriever');
+  }
+
+  async getEventIndexPattern(ctx: RequestHandlerContext) {
+    return await this.getIndexPattern(ctx, EndpointAppConstants.EVENT_DATASET);
+  }
+
+  async getMetadataIndexPattern(ctx: RequestHandlerContext) {
```

Comment:
would you mind adding a quick doc comment to public methods? maybe

```
/**
 * fetch and return the index pattern used to store Endpoint host metadata.
 */
```

I think these types of comments will help a lot, esp. considering the size of the org